### PR TITLE
Feature/oasim

### PR DESCRIPTION
**Pull request by travissluka**
## Description
Add initial code to `develop`. (I take zero credit for doing this!)

Required by:
- https://github.com/JCSDA-internal/ufo/pull/2339
- https://github.com/JCSDA-internal/ufo-bundle/pull/83

### Dependencies
None: can be merged at any time, does not require coordination.

### Testing 
when built as part of the ufo-bundle, the ctest `oasim_test` passes (with the below mentioned uncommitted change added)

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  data/atmo25b.dat
+  data/pic_sigma.dat
+  data/slingo.dat
+)
+
+# Create a directory to hold data files in the testing context
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/data/)
+
+# Link data to test directory
+LINK_FILES( "${data_files}" ${CMAKE_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
```
Comment (travissluka) [inline test/CMakeLists.txt]:
```suggestion
LINK_FILES( "${data_files}" ${CMAKE_CURRENT_SOURCE_DIR}/../ ${CMAKE_CURRENT_BINARY_DIR})
```

**Comment by travissluka** [comment]
@danholdaway, There is one bug fix to get the ctest to work. I don't have write permission on this repo so I can't push it. (and I can't request reviewers either :man_shrugging: )

**Comment by danholdaway** [comment]
@travissluka I just made you an admin

**Review by liuxiao37k — APPROVED** [review]
I swear that I reviewed and approved this PR Tuesday night... well, I built and tested with ufo-bundle. Checked the oasim fortran binaries and the dummy test_oasim.x. They all compiled fine.

**Pull request merged**